extension/httpfs/cache_functions.cpp

```diff
@@ -0,0 +1,83 @@
+#include "duckdb/function/function_set.hpp"
+#include "duckdb/parser/parsed_data/create_scalar_function_info.hpp"
+#include "httpfs-extension.hpp"
+#include "duckdb/main/client_data.hpp"
+#include "httpfs.hpp"
+#include "duckdb/common/http_state.hpp"
+
+namespace duckdb {
+
+void RegisterCacheFunction(duckdb::DataChunk &args, duckdb::ExpressionState &state, duckdb::Vector &result) {
+	D_ASSERT(args.ColumnCount() == 1);
```

Comment:
Maybe this should be a table function instead - so you can do `CALL register('...')` and we don't need to worry about running it over multiple rows?

Reply:
Good Point